tests/libwait/wait.go

```diff
@@ -39,25 +39,89 @@ import (
 	"kubevirt.io/kubevirt/tests/watcher"
 )
 
-// WaitForVMIStartOrFailed blocks until the specified VirtualMachineInstance reached either Failed or Running states
-func WaitForVMIStartOrFailed(obj runtime.Object, seconds int, wp watcher.WarningsPolicy) *v1.VirtualMachineInstance {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-	return waitForVMIPhase(ctx, []v1.VirtualMachineInstancePhase{v1.Running, v1.Failed}, obj, seconds, wp, true)
+const defaultTimeout = 360
+
+// Option represents an action that enables an option.
+type Option func(waiting Waiting)
+
+// Waiting represents the waiting struct.
+type Waiting struct {
+	ctx         context.Context
+	wp          *watcher.WarningsPolicy
+	timeout     int
+	waitForFail bool
+}
+
+// WaitingFor instantiates a Waiting configuration and return the vmi,
+// building its properties based on the specified With* options.
+func WaitingFor(obj runtime.Object, phases []v1.VirtualMachineInstancePhase, opts ...Option) *v1.VirtualMachineInstance {
```

Comment:
Thank you so much! I agree! I didn't want to change the signature of  `waitForVMIPhase` because I though that it was used in too many places. But from a deeper look it can be easily changed. Thanks, updates  will come soon :)